### DISCUSSION TITLE
Navigating without the navigation prop

### DISCUSSION THREAD

**Discussion by NandoMB** [expo/router#909] [Q&A]
Hello guys,

I'd like to know if anyone had the same issue as me... I'm trying to use the navigation "prop" inside my custom Axios interceptor. The problem is that I can't use a hook, as useNavigation, inside of it.
Using the React Navigation, you can have a "Navigation Service" based on ref. This is what I'm trying to do, but without success because we don't have a "NavigationContainer" component "exposed", right? How could I do that?

Here you can see the docs of [React Navigation](https://reactnavigation.org/docs/navigating-without-navigation-prop) showing how to deal with it. 

This is what I'm looking for, a way to link my 'navigationRef' to the expo-router 'NavigationContainer'...

```ts
import { StackActions, createNavigationContainerRef, ParamListBase } from '@react-navigation/native';

export const navigationRef = createNavigationContainerRef<ParamListBase>();

export function navigate(name: string, params?: object) {
  navigationRef.current?.navigate(name, params);
}

export function push(name: string, params?: object) {
  navigationRef.current?.dispatch(StackActions.push(name, params));
}

export function goBack() {
  navigationRef.current?.goBack();
}

```

**Comment by a-eid** [comment]
FYI to get access to the navigation ref, you have to use router-store

however this is not recommended and might not be supported for long.


```tsx
import { store } from "expo-router/build/global-state/router-store"

store.navigationRef
```
